app/livechat/server/business-hour/BusinessHourManager.ts

```diff
@@ -0,0 +1,110 @@
+import moment from 'moment';
+
+import { ILivechatBusinessHour } from '../../../../definition/ILivechatBusinessHour';
+import { ICronJobs } from '../../../utils/server/lib/cron/Cronjobs';
+import { IBusinessHour } from './AbstractBusinessHour';
+import { settings } from '../../../settings/server';
+
+const cronJobDayDict: Record<string, number> = {
+	Sunday: 0,
+	Monday: 1,
+	Tuesday: 2,
+	Wednesday: 3,
+	Thursday: 4,
+	Friday: 5,
+	Saturday: 6,
+};
+
+export class BusinessHourManager {
+	private businessHour: IBusinessHour;
+
+	private cronJobs: ICronJobs;
+
+	private cronJobsCache: string[] = [];
+
+	constructor(businessHour: IBusinessHour, cronJobs: ICronJobs) {
+		this.cronJobs = cronJobs;
+		this.registerBusinessHourMethod(businessHour);
```

Comment:
The compiler warns the `private businessHour` prop has no initializer because you're calling `registerBusinessHourMethod` instead of setting the default value to the prop. It's not a big deal, but the compiler is recognizing it.